awx_collection/plugins/modules/tower_instance_group.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+# coding: utf-8 -*-
+
+
+# (c) 2020, John Westcott IV <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_instance_group
+author: "John Westcott IV (@john-westcott-iv)"
+version_added: "4.0"
+short_description: create, update, or destroy Ansible Tower instance groups.
+description:
+    - Create, update, or destroy Ansible Tower instance groups. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name of this instance group.
+      required: True
+      type: str
+    new_name:
+      description:
+        - Setting this option will change the existing name (looked up via the name field.
+      required: True
+      type: str
+    credential:
+      description:
+        - Credential to authenticate with Kubernetes or OpenShift.  Must be of type "Kubernetes/OpenShift API Bearer Token”.
+      required: False
+      type: str
+    policy_instance_percentage:
+      description:
+        - Minimum percentage of all instances that will be automatically assigned to this group when new instances come online.
+      required: False
+      type: int
+      default: '0'
+    policy_instance_minimum:
+      description:
+        - Static minimum number of Instances that will be automatically assign to this group when new instances come online.
+      required: False
+      type: int
+      default: '0'
+    policy_instance_list:
+      description:
+        - List of exact-match Instances that will be assigned to this group
+      required: False
+      type: dict
+    pod_spec_override:
+      description:
+        - A custom Kubernetes or OpenShift Pod specification.
+      required: False
+      type: str
+    instances:
+      description:
+        - The instances associated with this instance_group
```

Comment:
We should get some info about the durability of this change. Does it remove instances, or just add them?

Reply:
Also, when will the policy rules undo this...

Reply:
It was kid of unclear what this does since there is also the "normal" association pattern as well.

Reply:
This just flows into our standard association pattern.